src/utils/cloud-backup/index.ts

```diff
@@ -0,0 +1,132 @@
+import { GoogleSignin } from '@react-native-google-signin/google-signin';
+import { isString } from 'lodash';
+import RNCloudFs from 'react-native-cloud-fs';
+import * as RNFS from 'react-native-fs';
+
+import { isAndroid } from 'src/config/system';
+import * as AesEncryption from 'src/utils/aes-encryption';
+
+import PackageJSON from '../../../package.json';
+
+export const cloudTitle = isAndroid ? 'Google Drive' : 'iCloud';
+
+const CLOUD_WALLET_FOLDER = 'temple-wallet';
+const filename = 'wallet-backup.json';
+const targetPath = `${CLOUD_WALLET_FOLDER}/${filename}`;
+
+export interface BackupFileInterface {
+  version: string;
+  mnemonic: string;
+}
+
+export const isCloudAvailable = () => {
+  if (isAndroid) {
+    return true;
+  }
+
+  return RNCloudFs.isAvailable();
+};
+
+export const requestSignInToCloud = async () => {
+  if (!isAndroid) {
+    return true;
+  }
+
+  try {
+    GoogleSignin.configure();
+
+    await GoogleSignin.signOut();
+    /* Syncing signed-in state to RNCloudFS */
+    await RNCloudFs.logout();
+
+    const user = await GoogleSignin.signIn();
+
+    if (user == null) {
+      return false;
+    }
+
+    /* Syncing signed-in state to RNCloudFS */
+    return await RNCloudFs.loginIfNeeded();
+  } catch (error) {
+    console.error(error);
+
+    return false;
+  }
+};
+
+export const fetchCloudBackupFileDetails = async () => {
+  const backups = await RNCloudFs.listFiles({
+    scope: 'hidden',
+    targetPath: CLOUD_WALLET_FOLDER
+  });
+
+  return backups.files?.find(file => file.name.endsWith(filename));
+};
+
+export const saveCloudBackup = async (mnemonic: string, password: string) => {
```

Comment:
This function seems too big, let's split it, please. For example: `getFileId` and `checkIfFileExists`